### ISSUE TITLE
Tex file taking very long to compile/render figures 

### ISSUE THREAD

**Issue by zoelzw** [milesdai/TAoE3Solutions#32]
As we added more figures to the tex file, I noticed that it took longer and longer to compile the document. I was trying to add a phasor diagram but it took too long. I didn't get the timeout error but it was time-consuming to have to wait to see the edits. I don't know if others have the same problem, but it'd be nice if we can improve the efficiency of compiling. We could divide each chapter into smaller chunks in different tex files. Or, would you consider organizing figures externally as described [here](https://www.overleaf.com/learn/latex/Questions/I_have_a_lot_of_tikz%2C_matlab2tikz_or_pgfplots_figures%2C_so_I%27m_getting_a_compilation_timeout._Can_I_externalise_my_figures%3F)? Thanks!

**Comment by milesdai** [comment]
Yeah this is a good point. I've integrated the `externalize` tikzpackage and worked out all the bugs to get things to compile. Take a look at #34 for the details.

Testing on my local machine (which admittedly is relatively slow), it takes the compile time from 11 seconds down to 6 seconds when the figures are already generated. On the downside, the initial compilation takes around 2.5 minutes.

I'm going to continue the discussion of the details of this implementation on #34, and I would recommend taking a look over there since this adds somewhat significant workflow changes.

**Comment by milesdai** [comment]
With all that being said, this solution still doesn't quite scale well as the number of chapters increases. If you are using overleaf, my suspicion is that we will hit the time limit for Overleaf again at some point especially since the project is currently mostly just Chapter 1.

When I first started this project, I had actually intended to keep each chapter separate because I was afraid of long compile times. I guess I forgot about that in the latest reorganization of the latex document (it seemed kind of nice to compile a single final document). Maybe something that's more sustainable is to use the `standalone` package to split the document by chapter so that each chapter can be compiled on its own.
